src/inmanta/agent/agent_new.py

```diff
@@ -132,8 +133,14 @@ def periodic_schedule(
                 return True
             return False
 
-        periodic_schedule("deploy", self.scheduler.deploy, self._deploy_interval, self._deploy_splay_value)
-        periodic_schedule("repair", self.scheduler.repair, self._repair_interval, self._repair_splay_value)
+        async def interval_deploy() -> None:
+            await self.scheduler.deploy(TaskPriority.INTERVAL_DEPLOY)
+
+        async def interval_repair() -> None:
+            await self.scheduler.repair(TaskPriority.INTERVAL_REPAIR)
```

Comment:
Hmm, I don't like this at all. I mean the fact that `mypy` is completely fine with the following, yet it gives an `AttributeError`:

```python
import functools
from collections.abc import Callable


def f() -> int:
    return 42


x: Callable[[], int] = functools.partial(f)
print(x.__name__)
```

But given those circumstances, I agree with your assessment.

Reply:
We could also make the scheduler more robust by trying `__name__` and falling back to `repr(func)`, but since that's out of scope for this ticket I'll leave the decision up to you.